server/models/metadata.model.js

```diff
@@ -135,21 +141,76 @@ MetadataSchema.statics = {
 
       if (wiki) searchQuery.wiki = wiki
 
-      return this.find(searchQuery)
-        .skip(+start)
-        .limit(+end)
-        .sort({ score: 'desc' })
-        .exec()
-        .then((metadata) => {
-
-          if (search) {
-            return metadata.map(item => item._id)
-          } else {
-            return metadata.map(item => item)
-          }
+      if (discover) {
+        // find each discover item separately (promise reduce)
+        let allDiscoverItems = [[],[]]
+        return discovers.reduce(
+          (p, typeToSearch) => p.then(() => {
+            if (typeToSearch === "e") {
+              delete searchQuery.subtype
+              searchQuery.type = typeToSearch
+            } else {
+              searchQuery.type = "i"
+              searchQuery.subtype = typeToSearch
+            }
+            return this.find(searchQuery)
+              .skip(+start)
+              .limit(+end)
+              .sort({ score: 'desc' })
+              .exec()
+              .then((metadata) => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to not always return the same value. ([javascript:S3516](https://sonarcloud.io/organizations/aumanjoa-github/rules#rule_key=javascript%3AS3516))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=chronas-api&pullRequest=29&issues=AWbY54ufSgN4uCclP4pk&open=AWbY54ufSgN4uCclP4pk)

[ISSUE_KEY]: AWbY54ufSgN4uCclP4pk